docs/src/main/asciidoc/kafka.adoc

```diff
@@ -3,438 +3,946 @@ This guide is maintained in the main Quarkus repository
-However, you can go right to the completed example.
+You can add the `smallrye-reactive-messaging-kafka` extensions to your project by running the following command in your project base directory:
 
-Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+[source,bash]
+----
+./mvnw quarkus:add-extension -Dextensions="smallrye-reactive-messaging-kafka"
+----
+
+This will add the following to your `pom.xml`:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-smallrye-reactive-messaging-kafka</artifactId>
+</dependency>
+----
 
-The solution is located in the `kafka-quickstart` {quickstarts-tree-url}/kafka-quickstart[directory].
+== Configuring Smallrye Kafka Connector
 
-== Creating the Maven Project
+Because Smallrye Reactive Messaging framework supports different messaging backends like Apache Kafka, AMQP, Apache Camel, JMS, MQTT, etc., it employs a generic vocabulary:
 
-First, we need a new project. Create a new project with the following command:
+- Applications send and receive *messages*. A message wraps a _payload_ and can be extended with some _metadata_. With the Kafka connector, a _message_ corresponds to a Kafka _record_.
+- Messages transit on *channels*. Application components connect to channels to publish and consume messages. The Kafka connector maps _channels_ to Kafka _topics_.
+- Channels are connected to message backends using *connectors*. Connectors are configured to map incoming messages to a specific channel (consumed by the application) and collect outgoing messages sent to a specific channel. Each connector is dedicated to a specific messaging technology. For example, the connector dealing with Kafka is named `smallrye-kafka`.
 
-[source,bash,subs=attributes+]
+A minimal configuration for the Kafka connector with an incoming channel looks like the following:
+
+[source, properties]
 ----
-mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
-    -DprojectGroupId=org.acme \
-    -DprojectArtifactId=kafka-quickstart \
-    -DclassName="org.acme.kafka.PriceResource" \
-    -Dpath="/prices" \
-    -Dextensions="resteasy-reactive,smallrye-reactive-messaging-kafka"
-cd kafka-quickstart
+kafka.bootstrap.servers=kafka:9092 <1>
+mp.messaging.incoming.prices.connector=smallrye-kafka <2>
 ----
 
-This command generates a Maven project, importing the Reactive Messaging and Kafka connector extensions.
+<1> Configure the broker location. You can configure it globally or per channel. It defaults to localhost:9092
+<2> Configure the connector to manage the prices channel. By default the topic name is same as the channel name. You can configure the topic attribute to override it.
 
-If you already have your Quarkus project configured, you can add the `smallrye-reactive-messaging-kafka` extension
-to your project by running the following command in your project base directory:
+== Receiving messages from Kafka
```

Comment:
If you remove the bootstrap servers below add a note here indicating that is can be configured globally or per channel and mention dev services.